File: app/src/main/java/org/doomer/qnotez/db/DateConverter.java
package org.doomer.qnotez.db;

import android.arch.persistence.room.TypeConverter;
import java.util.Date;

public class DateConverter {

    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? 0 : date.getTime();
    }
}

File: app/src/main/kotlin/org/doomer/qnotez/db/DateConverter.kt
package org.doomer.qnotez.db

import android.arch.persistence.room.TypeConverter
import java.util.Date

class DateConverter {
    @TypeConverter  // Указать, что метод является конвертером
    fun toDate(timestamp: Long?) = timestamp?.let { Date(it) }

    @TypeConverter  // Указать, что метод является конвертером
    fun toTimestamp(date: Date?) = date?.time
}
